### PR TITLE
Add support for postgresql JSON type

### PR DESCRIPTION
**Pull request by supherman**

### CODE REVIEW

File: json/json.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+)
+
+type JSON struct {
+	Raw json.RawMessage
+}
+
+func (jv *JSON) Scan(value interface{}) error {
```
Comment (echlebek) [inline json/json.go]: How about io.Reader here?

File: json/json_test.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	jsonValue := JSON{}
+
+	// Test null values
+	err = db.QueryRow("SELECT null::json").Scan(&jsonValue)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if jsonValue.Raw != nil {
+		t.Fatalf("expected empty value")
```
Comment (cbandy) [inline json/json_test.go]: `Fatal ` here.

**Comment by cbandy** [comment]
I'm not sure I understand what new feature this enables. I am able to scan into a `json.RawMessage` already (without this code.) The struct seems like unnecessary indirection.

I haven't personally tried using `json.RawMessage` as a Value, but I don't see anything here that... does anything?

**Comment by supherman** [comment]
@cbandy you know what, you're right, I misunderstood how the scan method works, I can do the same thing just passing a []byte value and it's going to work, the good part is that lib/pq supports postgresql's json type out of the box, anyway I will close this, thanks for the quick feedback, fortunately this just took me 30 minutes.

**Comment by cbandy** [comment]
I personally think it may still be worthwhile to have some tests that ensure/show `json.RawMessage` is a viable Value.

It seems like "the" type to represent... arbitrary JSON.

**Pull request closed without merging**